### PR TITLE
Added support for handling SAML2 assertions already wrapped in respon…

### PR DESCRIPTION
**Pull request by cmoesgaard**
…ses (as is the case with the SSO response)

### CODE REVIEW

File: oio_rest/oio_rest/auth/saml2.py [hunk 1]
```diff
@@ -34,15 +34,22 @@ def __init__(self, assertion_xml, mox_entity_id, idp_entity_id, idp_url,
...
-        # so we fake it here.
-        self.response = """
-        <samlp:Response xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol"
-                        xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion">
-            %s
-        </samlp:Response>""" % assertion_xml
-
-        self.original_document = fromstring(self.response)
+        document = fromstring(assertion_xml)
+        if 'Response' not in document.tag:
```
Comment (danchr) [inline oio_rest/oio_rest/auth/saml2.py]: Any particular reason not do a direct comparison?

File: oio_rest/oio_rest/auth/saml2.py [hunk 2]
```diff
@@ -34,15 +34,22 @@ def __init__(self, assertion_xml, mox_entity_id, idp_entity_id, idp_url,
...
+            # OneLogin's methods expect the data wrapped in a Response element,
+            # so we fake it here.
+            response = """
+            <samlp:Response xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol"
+                            xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion">
+                %s
+            </samlp:Response>""" % assertion_xml
```
Comment (agger-magenta) [inline oio_rest/oio_rest/auth/saml2.py]: I think we should use `format()` for string formatting, it's more powerful and easier to read.

Comment (danchr) [reply oio_rest/oio_rest/auth/saml2.py]: In @cmoesgaard's defence, he merely copied this code from elsewhere 🙂 In this case, though, I'd say that `''.join(…)` is customary, given that the code isn't doing any actual formatting.

Comment (agger-magenta) [reply oio_rest/oio_rest/auth/saml2.py]: Whatever! I'd like to deprecate the `%` operator for string formatting.

Comment (agger-magenta) [reply oio_rest/oio_rest/auth/saml2.py]: And in Python 3.6 we'll get these cute `f"..."` thingies.

File: oio_rest/oio_rest/auth/saml2.py (hunk 1 above)
Comment (cmoesgaard) [reply oio_rest/oio_rest/auth/saml2.py]: `document.tag` contains the tag's namespace as well, e.g. `'{urn:oasis:names:tc:SAML:2.0:protocol}Response'`

As an alternative, we could so some simple regex matching to extract the part after the `{}` part, and do our comparison on that.

File: oio_rest/oio_rest/auth/saml2.py (hunk 2 above)
Comment (cmoesgaard) [reply oio_rest/oio_rest/auth/saml2.py]: Agreed, I'll fix that.

File: oio_rest/oio_rest/auth/saml2.py (hunk 1 above)
Comment (danchr) [reply oio_rest/oio_rest/auth/saml2.py]: Gah, and there's no way to access the unprefixed tag, or add a namespace? (I remember running into this earlier, and yeah, it _is_ bloody annoying…)

Comment (cmoesgaard) [reply oio_rest/oio_rest/auth/saml2.py]: No way of getting _just_ the tag name, as far as I can tell.

Comment (cmoesgaard) [reply oio_rest/oio_rest/auth/saml2.py]: It probably isn't going to change, no. You're right.

Comment (agger-magenta) [reply oio_rest/oio_rest/auth/saml2.py]: I think either that or add a comment on why we're doing it like this.

File: oio_rest/oio_rest/auth/saml2.py [hunk 3]
```diff
@@ -35,7 +35,7 @@ def __init__(self, assertion_xml, mox_entity_id, idp_entity_id, idp_url,
         self.idp_cert = idp_cert
 
         document = fromstring(assertion_xml)
-        if 'Response' not in document.tag:
+        if document.tag != '{urn:oasis:names:tc:SAML:2.0:protocol}Response':
```
Comment (danchr) [inline oio_rest/oio_rest/auth/saml2.py]: This still feels vaguely horrible… that's definitely not your fault, though, but rather a case of weird APIs for a horrible format.

**Pull request merged**